wordpress-mobile/WordPress-iOS
# Post Scheduling: adding new flow for setting publish date and time

Ref #12686 
Ref PRs:
- https://github.com/wordpress-mobile/WordPress-iOS/pull/12749
- https://github.com/wordpress-mobile/WordPress-iOS/pull/12768
- https://github.com/wordpress-mobile/WordPress-iOS/pull/12838
- https://github.com/wordpress-mobile/WordPress-iOS/pull/12851
- https://github.com/wordpress-mobile/WordPress-iOS/pull/12860
- https://github.com/wordpress-mobile/WordPress-iOS/pull/12879

This encompasses all @bjtitus' changes that adds a new flow for setting a Post's publish date and time.

To test:

---
- Start a new post.
- Tap the ellipsis in the nav bar, and select 'Post Settings'.
- Select 'Publish'.
- Verify the new 'Publish' view is displayed, with a 'Date and Time' row.

![init](https://user-images.githubusercontent.com/1816888/67969572-2ccdd600-fbcf-11e9-9893-fa9ebb06e9e2.png)

---
- Select 'Date and Time'.
- A calendar is displayed, with the ability to:
  - Set to 'Publish immediately'.
  - Select a date, and 'Next' to continue.
  - When 'Next' is selected, a time picker is displayed.

iPhone portrait:

|  |  |
|--------|-------|
|![date](https://user-images.githubusercontent.com/1816888/67969637-4d962b80-fbcf-11e9-8b1e-88e5168b95c6.png)|![time](https://user-images.githubusercontent.com/1816888/67969651-54bd3980-fbcf-11e9-9154-54b0351a6cb9.png)|

iPhone landscape:

|  |  |
|--------|-------|
|![landscape_date](https://user-images.githubusercontent.com/1816888/67970385-82ef4900-fbd0-11e9-8da0-cb1421d81321.png)|![landscape_time](https://user-images.githubusercontent.com/1816888/67970399-87b3fd00-fbd0-11e9-839a-cd6eeb9b8f29.png)|

iPad:

|  |  |
|--------|-------|
|![ipad_date](https://user-images.githubusercontent.com/1816888/67970135-1ffdb200-fbd0-11e9-9582-05e676c39bb2.png)|![ipad_time](https://user-images.githubusercontent.com/1816888/67970151-25f39300-fbd0-11e9-884a-8839f27af246.png)|

---
- After selecting a date and time, it will be displayed on the 'Publish' view.

![after](https://user-images.githubusercontent.com/1816888/67969789-8fbf6d00-fbcf-11e9-8f30-f2d600e0003f.png)

---
- Disable `postScheduling` feature flag.
- Verify the old date picker appears when 'Publish' is selected.

![old](https://user-images.githubusercontent.com/1816888/67969799-93eb8a80-fbcf-11e9-9528-ef858fe918d5.png)

---
Update release notes:
